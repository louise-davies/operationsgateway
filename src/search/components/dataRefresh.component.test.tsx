import React from 'react';
import DataRefresh, { type DataRefreshProps } from './dataRefresh.component';
import { render, screen, type RenderResult } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('DataRefresh', () => {
  let props: DataRefreshProps;
  const refreshData = jest.fn();

  const createView = (): RenderResult => {
    return render(<DataRefresh {...props} />);
  };

  beforeEach(() => {
    props = {
      timeframeSet: true,
      refreshData,
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { asFragment } = createView();

    expect(asFragment()).toMatchSnapshot();
  });

  it('is disabled and does not call refreshData on click if no timeframe is currently set', async () => {
    props.timeframeSet = false;
    createView();

    const button = screen.getByRole('button', { name: 'Refresh data' });
    expect(button).toBeDisabled();
  });

  it('calls refreshData on click if timeframe set', async () => {
    const user = userEvent.setup();
    createView();

    const button = screen.getByRole('button', { name: 'Refresh data' });
    await user.click(button);
    expect(refreshData).toHaveBeenCalledTimes(1);
  });
});
