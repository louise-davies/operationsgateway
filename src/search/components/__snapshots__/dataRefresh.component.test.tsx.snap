// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataRefresh renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-12b4m5q"
  >
    <button
      aria-label="Refresh data"
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RefreshIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4z"
          />
        </svg>
      </span>
      Refresh
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
